internal/httpproxy/proxy.go

```diff
@@ -0,0 +1,25 @@
+package httpproxy
+
+import (
+	"fmt"
+	"net/http"
+)
+
+// Proxy definition
+type Proxy map[string]Store
+
+// NewProxy allocates a new http proxy from its configuration
+func NewProxy(cfg Proxy) *Proxy {
+	proxy := new(Proxy)
```

Comment:
The config is the same as the actual struct we are working on (the proxy?). This is weird and should be actually two different structs. One for the configuration, one for the "server" struct.